crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -45,6 +45,9 @@ pub(super) struct SemanticIndexBuilder<'db> {
     /// Flow states at each `break` in the current loop.
     loop_break_states: Vec<FlowSnapshot>,
 
+    /// Flags about the file's global scope
+    has_future_annotations: bool,
```

Comment:
We probably want to make this a bit flag long term but a boolean is just fine for now :)